app/src/main/java/org/p2p/wallet/common/mvp/BaseFragment.kt

```diff
@@ -70,20 +73,26 @@ abstract class BaseFragment(@LayoutRes layoutRes: Int) : Fragment(layoutRes), Ba
         arguments = (arguments ?: Bundle()).apply { putInt(extraKey, animation) }
     }
 
-    private fun updateStatusBarColor() {
-        requireActivity().window.statusBarColor = resources.getColor(
-            statusBarColor,
-            requireActivity().theme
-        )
+    private fun setStatusBarColor(@ColorRes colorResId: Int) {
+        val window = requireActivity().window ?: return
+        with(window) {
+            addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
+            statusBarColor = resources.getColor(colorResId, requireActivity().theme)
+        }
+        WindowCompat.setDecorFitsSystemWindows(window, false)
     }
 
-    private fun setStatusBarIconsColor(dark: Boolean) {
-        val decor: View = requireActivity().window.decorView
-        if (dark) {
-            decor.systemUiVisibility = View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
-        } else {
-            decor.systemUiVisibility = 0
-        }
+    /*
+        Change status bar icon color
+        isLight = true - dark color
+        isLight = false - white color
+
+        Don't forget to reset status bar color on fragment destroy to restore previous color
+     */
+    protected fun setLightStatusBar(isLight: Boolean) {
```

Comment:
maybe we should find another name than `isLight` that removes kdoc need?
or are we changing status bar icon color? so maybe we should rename a whole method?
`setStatusBarIconsList(makeIconsLight: Boolean)`

Reply:
it would be better to make kdoc this way:
```
    /*
        Change status bar and its icons color
        isLight = true - white status bar, dark icons
        isLight = false - dark status bar, white icons
        Don't forget to reset status bar color on fragment destroy to restore previous color
     */
```